students/jim_krecek/lesson06/test_mailroom_p4.py

```diff
@@ -0,0 +1,57 @@
+import mailroom_p4
+from mailroom_p4 import *
+import os
+import pytest
+
+def test_create_report():
+    report = """Donor Name               | Total Given | Num Gifts | Average Gift
+-----------------------------------------------------------------
+Pat Carrier               $     106.00           2  $       53.00
+Jim Krecek                $      93.00           3  $       31.00
+Nate Secinaro             $      65.00           3  $       21.67
+Zac Fisher                $      21.00           3  $        7.00
+Jess Reid                 $       3.00           1  $        3.00"""
+    assert mailroom_p4.create_report() == report
+    
+
+def test_list():
+    data = {"Nate Secinaro":[12,44,9],"Jess Reid":[3],"Pat Carrier":[65,41]}
+    assert mailroom_p4.donor_list(data) == """Nate Secinaro
+Jess Reid
+Pat Carrier
+"""
+
+def test_add_update():
+    # tests the addition of new donors, update of existing donors, and sending of 
+    # thank you message
+    
+    # tests addition of new donor
+    data = {"Nate Secinaro":[12,44,9],"Jess Reid":[3],"Pat Carrier":[65,41]}
+    assert mailroom_p4.thanks(data, 'joe', 44) == """
+Dear joe,
+
+Thank you for your generous donation of $44.
+
+Sincerely, Jim"""
+    
+    # tests update of current donor
+    assert mailroom_p4.thanks(data, 'Nate Secinaro', 39) == """
+Dear Nate Secinaro,
+
+Thank you for your generous donation of $39.
+
+Sincerely, Jim"""
+
+def letter_test():
+    # test the amount of letters generated
+    data = {"Nate Secinaro":[12,44,9],"Jess Reid":[3],"Pat Carrier":[65,41]}
+    for k,v in data.items:
+        mailroom_p4.write_letters(k,v)
```

Comment:
since you're only testing `write_letters()` function, you shouldn't actually call it in a loop.

you can execute it once with a donor and assert that a single file was created.